Mathlib/CategoryTheory/Adjunction/Basic.lean

```diff
@@ -176,6 +176,18 @@ theorem homEquiv_naturality_right_symm (f : X ⟶ G.obj Y) (g : Y ⟶ Y') :
   simp only [homEquiv_naturality_right,eq_self_iff_true,Equiv.apply_symm_apply]
 #align category_theory.adjunction.hom_equiv_naturality_right_symm CategoryTheory.Adjunction.homEquiv_naturality_right_symm
 
+theorem homEquiv_naturality_left_square (f : X' ⟶ X) (g : F.obj X ⟶ Y')
+    (h : F.obj X' ⟶ Y) (k : Y ⟶ Y') (w : F.map f ≫ g = h ≫ k) :
+    f ≫ (adj.homEquiv X Y') g = (adj.homEquiv X' Y) h ≫ G.map k := by
+  rw [← homEquiv_naturality_left, ← homEquiv_naturality_right]
+  exact congrArg (adj.homEquiv X' Y') w
+
+theorem homEquiv_naturality_right_square (f : X' ⟶ X) (g : X ⟶ G.obj Y')
+    (h : X' ⟶ G.obj Y) (k : Y ⟶ Y') (w : f ≫ g = h ≫ G.map k) :
+    F.map f ≫ (adj.homEquiv X Y').symm g = (adj.homEquiv X' Y).symm h ≫ k := by
+  rw [← homEquiv_naturality_left_symm, ← homEquiv_naturality_right_symm]
+  exact congrArg (adj.homEquiv X' Y').symm w
```

Comment:
```suggestion
  rw [← homEquiv_naturality_left_symm, ← homEquiv_naturality_right_symm, w]
```

Reply:
Right, of course! I'm not used to using rewrites, and forget how powerful this tactic is. Thanks.